// @sentry/electron/dist/dispatch.js
Object.defineProperty(exports, "__esModule", { value: true });
const tslib_1 = require("tslib");
const utils_1 = require("@sentry/utils");
/**
 * The Sentry Electron SDK Frontend.
 *
 * @see ElectronOptions for documentation on configuration options.
 */
class ElectronClient {
    /**
     * Creates a new Electron SDK instance.
     *
     * This constructor automatically chooses the right implementation for the
     * process type (`browser` or `renderer`).
     *
     * @param options Configuration options for this SDK.
     */
    constructor(options) {
        // We dynamically load the frontend implementation for the current process
        // type. In frontend bundlers such as webpack or rollup, those requires are
        // resolved statically. For this reason, we use `dynamicRequire` for the
        // main implementation here, which is only defined in the main process. The
        // renderer implementation must use the default `require`.
        // In case `process.type` is not defined, dispatch defaults to the renderer
        // implementation, which should be fine for most cases. False positives of
        // this would be running `@sentry/electron` in a bare node process, which is
        // acceptable.
        // tslint:disable:no-unsafe-any
        const clientClass = process.type === 'browser' ? utils_1.dynamicRequire(module, './main').MainClient : require('./renderer').RendererClient;
        // tslint:enable:no-unsafe-any
        this._inner = new clientClass(options);
    }
    /**
     * @inheritDoc
     */
    captureMinidump(path, event, scope) {
        return this._inner.captureMinidump(path, event, scope);
    }
    /**
     * @inheritDoc
     */
    captureException(exception, hint, scope) {
        return this._inner.captureException(exception, hint, scope);
    }
    /**
     * @inheritDoc
     */
    captureMessage(message, level, hint, scope) {
        return this._inner.captureMessage(message, level, hint, scope);
    }
    /**
     * @inheritDoc
     */
    captureEvent(event, hint, scope) {
        return this._inner.captureEvent(event, hint, scope);
    }
    /**
     * @inheritDoc
     */
    getDsn() {
        return this._inner.getDsn();
    }
    /**
     * @inheritDoc
     */
    getOptions() {
        return this._inner.getOptions();
    }
    /**
     * @inheritDoc
     */
    close(timeout) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            return this._inner.close(timeout);
        });
    }
    /**
     * @inheritDoc
     */
    flush(timeout) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            return this._inner.flush(timeout);
        });
    }
    /**
     * @inheritDoc
     */
    showReportDialog(options) {
        // tslint:disable-next-line
        this._inner.showReportDialog(options);
    }
    /**
     * @inheritDoc
     */
    getIntegration(integration) {
        return this._inner.getIntegration(integration);
    }
}
exports.ElectronClient = ElectronClient;
/**
 * This either calls init on main with node interations or renderer
 * with browser integrations.
 *
 * @param options Options
 */
function specificInit(options) {
    // tslint:disable-next-line
    process.type === 'browser' ? utils_1.dynamicRequire(module, './main').init(options) : require('./renderer').init(options);
}
exports.specificInit = specificInit;
/** Return all integrations depending if running in browser or renderer. */
function getIntegrations() {
    // tslint:disable:no-unsafe-any
    return process.type === 'browser'
        ? {
            electron: utils_1.dynamicRequire(module, './main').ElectronIntegrations,
            node: utils_1.dynamicRequire(module, './main').NodeIntegrations,
        }
        : { browser: require('./renderer').BrowserIntegrations };
    // tslint:enable:no-unsafe-any
}
exports.getIntegrations = getIntegrations;
