// @sentry/electron/dist/index.js
Object.defineProperty(exports, "__esModule", { value: true });
var types_1 = require("@sentry/types");
exports.Severity = types_1.Severity;
exports.Status = types_1.Status;
var core_1 = require("@sentry/core");
exports.addGlobalEventProcessor = core_1.addGlobalEventProcessor;
exports.addBreadcrumb = core_1.addBreadcrumb;
exports.captureException = core_1.captureException;
exports.captureEvent = core_1.captureEvent;
exports.captureMessage = core_1.captureMessage;
exports.configureScope = core_1.configureScope;
exports.getHubFromCarrier = core_1.getHubFromCarrier;
exports.getCurrentHub = core_1.getCurrentHub;
exports.Hub = core_1.Hub;
exports.Scope = core_1.Scope;
exports.setContext = core_1.setContext;
exports.setExtra = core_1.setExtra;
exports.setExtras = core_1.setExtras;
exports.setTag = core_1.setTag;
exports.setTags = core_1.setTags;
exports.setUser = core_1.setUser;
exports.withScope = core_1.withScope;
var common_1 = require("./common");
exports.CommonBackend = common_1.CommonBackend;
var dispatch_1 = require("./dispatch");
exports.ElectronClient = dispatch_1.ElectronClient;
exports.getIntegrations = dispatch_1.getIntegrations;
var sdk_1 = require("./sdk");
exports.captureMinidump = sdk_1.captureMinidump;
exports.init = sdk_1.init;
exports.showReportDialog = sdk_1.showReportDialog;
